topics/dev/tutorials/interactive-tools/tutorial.md

```diff
@@ -525,6 +530,535 @@ our new Docker container as a Galaxy tool.
+
+```
+
+You can also add your own set of notebooks to guide the user like that :
+
+```
+# Download notebooks
+RUN cd /home/$NB_USER/ &&  \
+    wget -O notebook-samples.zip https://github.com/eu-cdse/notebook-samples/archive/refs/heads/main.zip && \
+    unzip notebook-samples.zip && \
+    rm /home/$NB_USER/notebook-samples.zip && \
+    mv /home/$NB_USER/notebook-samples-main/geo /home/$NB_USER && \
+    mv /home/$NB_USER/notebook-samples-main/sentinelhub /home/$NB_USER && \
+    mv /home/$NB_USER/notebook-samples-main/openeo /home/$NB_USER && \
+    rm -r /home/$NB_USER/notebook-samples-main
+```
+
+Finally, some general variables of environment :
+
+```
+# ENV variables to replace conf file
+ENV DEBUG=false \
+    GALAXY_WEB_PORT=10000 \
+    NOTEBOOK_PASSWORD=none \
+    CORS_ORIGIN=none \
+    DOCKER_PORT=none \
+    API_KEY=none \
+    HISTORY_ID=none \
+    REMOTE_HOST=none \
+    GALAXY_URL=none
+
+# @jupyterlab/google-drive  not yet supported
+
+USER root
+WORKDIR /import
+
+# Start Jupyter Notebook
+CMD /startup.sh
+
+```
+
+And with all this your Dockerfile is ready 
+
+You now need to publish this image o a public repository.
+Anyone can use this Dockerfile to rebuild the image if necessary.
+In any case don't forget to have all the files shown when we cloned this [repository](https://github.com/bgruening/docker-jupyter-notebook/tree/master) next to your Dockerfile.
+
+> <hands-on-title></hands-on-title>
+>
+> Let's start working on this Docker container.
+>
+> 1. Install Docker as described on the [docker website](https://docs.docker.com/engine/install/). Click on your distribution name to get specific information.
+>
+> 2. Now let's use the recipe to build our Docker image.
+>
+>    ```sh
+>    # Build a container image from our Dockerfile
+>    IMAGE_TAG="myimage"
+>    LOG_PATH=`pwd`  # Create log output in current directory
+>    PORT=8765
+>    docker build -t $IMAGE_TAG --build-arg LOG_PATH=$LOG_PATH --build-arg PORT=$PORT .
+>    ```
+>
+>    > <tip-title>Automating the build</tip-title>
+>    > While developing the Docker container you may find yourself tweaking and rebuilding the container image many times.
+>    > In the GitHub repository linked above, you'll notice that the author has used a `Makefile` to accelerate the build and deploy process.
+>    > This allows the developer to simply run `make docker` and `make push_hub` to build and push the container, or `make` to rebuild the container after making changes during development. Check out the `Makefile` to see what commands can be run using `make` in this repository.
+>    >
+>    {: .tip}
+{: .hands_on}
+
+If you are lucky, you might find an available Docker image for the application you are trying to wrap. However, existing Docker images often require some "tweaking" before they will work as a GxIT. Some example configuration changes are:
+
+1. Expose the correct port. The application, Docker and tool XML ports must be aligned!
```

Comment:
Not sure if this is really needed to be honest.

Reply:
Which part ? thepart on tweaking the Docker image or the ports ?

Reply:
Tweaking the docker image.

Reply:
Better ?